In [1]:
import sympy as sp
from scipy import special

# Problem 1

In [4]:
special.binom(6+3, 6)

84.0

$84$ (E)

# Problem 2

In [5]:
s = 0
for k in 2, 3, 4:
    s += special.comb(k, 2, exact=True) * special.comb(9-k, 1, exact=True)
s

55

In [6]:
special.comb(9, 3, exact=True)

84

In [7]:
sp.Rational(55,84)

55/84

In [8]:
55 + 84

139

$139$

# Problem 3

$1$ way to sum to $2$. $2$ ways to sum to $3$. $3$ ways to sum to $4$. $4$ ways to sum to $5$. $5$ ways to sum to $6$.

In [2]:
3 * (1 + 2 + 3 + 4 + 5) / sp.sympify(6*6*6)

5/24

(D)

# Problem 4

Choose the center first. Then there is a (unique) set of $2$-element sets that represent which pairs of vertices that get what number if and only if $45$ minus the center is divisible by $4$. 

In [3]:
3 * 24 * 2**4

1152

$1152$ (C)

# Problem 5

For every $b$, there is two choices for $a$ and two choices for $c$.

In [14]:
10 * 2 * 2

40

$40$

# Problem 6

In [5]:
6*5*1*4*5

600

In [6]:
6*5*4*1*5

600

In [7]:
6*5*4*4*4

1920

In [8]:
600 + 600 + 1920

3120

$3120$ (C)

# Problem 7

$abc + ab + a \equiv a(bc + b + 1)\equiv 0\pmod 3$.
* $a\equiv 0\pmod 3$
* $bc + b + 1\equiv 0\pmod 3\implies (b+1)c\equiv 2\pmod 3\implies (b\equiv 0\pmod 3\land c\equiv 2\pmod 3)\lor(b\equiv 1\pmod 3 \land c\equiv 1\pmod 3)$

In [18]:
sp.Rational(1,3) + sp.Rational(2,3) * (sp.Rational(1,3)**2*2)

13/27

(E)

# Problem 8

In [19]:
1000 % 7, 100 % 7, 10%7

(6, 2, 3)

$0\equiv 1000a + 100b + 10b + a\equiv 6a + 2b + 3b + a\equiv 7a + 5b \equiv 5b\pmod 7\iff b\equiv 0\pmod 7$

In [4]:
num_palindromes = 9*10

In [5]:
num_good = 9 * 2

In [6]:
sp.Rational(num_good, num_palindromes)

1/5

(E)

# Problem 9

If first person picks a $9$ it's game over.

Otherwise, the probability that they tie is $\frac{1}{\binom83}$, so the probability that first person wins is $\frac12\cdot\left(1-\frac{1}{\binom 83}\right)$

In [33]:
nopicknine =(1- sp.Rational(1, special.comb(8, 3, exact=True)))/2
nopicknine

55/112

In [34]:
sp.Rational(1,3) + sp.Rational(2,3) * _

37/56

(B)

# Problem 10

In [36]:
from sympy.abc import p, q

In [40]:
sp.Eq(sp.Rational(1,6), 6*p**2*(1-p)**2)

Eq(1/6, 6*p**2*(1 - p)**2)

$\frac16 = p(1-p)$

In [42]:
sp.Eq(sp.Rational(1,6), p*(1-p))

Eq(1/6, p*(1 - p))

In [44]:
sp.solveset(_,p)

FiniteSet(1/2 - sqrt(3)/6, sqrt(3)/6 + 1/2)

$\frac{3-\sqrt3}{6}$ (D)